`mvn` failed for me locally with the following log, I don't think it's related to my change though.

```
[WARNING] /home/weih/sources/commons-configuration/target/generated-sources/javacc/org/apache/commons/configuration2/plist/SimpleCharStream.java:31: warning: no comment
[WARNING] protected int maxNextCharInd = 0;
[WARNING] ^
[WARNING] /home/weih/sources/commons-configuration/target/generated-sources/javacc/org/apache/commons/configuration2/plist/SimpleCharStream.java:25: warning: no comment
[WARNING] protected boolean prevCharIsCR = false;
[WARNING] ^
[WARNING] /home/weih/sources/commons-configuration/target/generated-sources/javacc/org/apache/commons/configuration2/plist/SimpleCharStream.java:26: warning: no comment
[WARNING] protected boolean prevCharIsLF = false;
[WARNING] ^
[WARNING] 100 warnings
[ERROR] Error while creating javadoc report: Project contains Javadoc Warnings
org.apache.maven.reporting.MavenReportException: Project contains Javadoc Warnings
    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.doExecuteJavadocCommandLine (AbstractJavadocMojo.java:5146)
    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.executeJavadocCommandLine (AbstractJavadocMojo.java:4987)
    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.executeReport (AbstractJavadocMojo.java:2059)
    at org.apache.maven.plugins.javadoc.JavadocReport.generate (JavadocReport.java:142)
    at org.apache.maven.plugins.javadoc.JavadocReport.doExecute (JavadocReport.java:307)
    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.execute (AbstractJavadocMojo.java:1851)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:919)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:285)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:207)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:362)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
```

--- a/src/main/java/org/apache/commons/configuration2/io/CombinedLocationStrategy.java
+++ b/src/main/java/org/apache/commons/configuration2/io/CombinedLocationStrategy.java
@@ -21,6 +21,7 @@ import java.net.URL;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.List;
 import java.util.Set;
 import java.util.regex.Pattern;
 
@@ -119,10 +120,11 @@ public class CombinedLocationStrategy extends AbstractFileLocationStrategy {
         if (builder.subStrategies == null) {
             throw new IllegalArgumentException("Collection with sub strategies must not be null.");
         }
-        if (builder.subStrategies.contains(null)) {
+        List<FileLocationStrategy> subStrategiesCopy = new ArrayList<>(builder.subStrategies);
+        if (subStrategiesCopy.contains(null)) {
             throw new IllegalArgumentException("Collection with sub strategies contains null entry.");
         }
-        subStrategies = Collections.unmodifiableCollection(new ArrayList<>(builder.subStrategies));
+        subStrategies = Collections.unmodifiableCollection(subStrategiesCopy);
     }
 
     /**

--- a/src/test/java/org/apache/commons/configuration2/io/TestCombinedLocationStrategy.java
+++ b/src/test/java/org/apache/commons/configuration2/io/TestCombinedLocationStrategy.java
@@ -26,9 +26,11 @@ import static org.mockito.Mockito.verifyNoMoreInteractions;
 import static org.mockito.Mockito.when;
 
 import java.net.URL;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.LinkedList;
+import java.util.Objects;
 
 import org.apache.commons.configuration2.ConfigurationAssert;
 import org.junit.jupiter.api.BeforeAll;
@@ -121,6 +123,25 @@ public class TestCombinedLocationStrategy {
         final Collection<FileLocationStrategy> col = new LinkedList<>(Arrays.asList(getSubStrategies()));
         col.add(null);
         assertThrows(IllegalArgumentException.class, () -> new CombinedLocationStrategy(col));
+    }
+
+    /**
+     * Tests that the constructor handles collections that throw NPE on contains(null) (like ImmutableList).
+     */
+    @Test
+    void testInitCollectionThrowsNPEOnContainsNull() {
+        // Create a collection that throws NPE on contains(null) like List.of() instance does
+        final Collection<FileLocationStrategy> collectionThatThrowsNPE = new ArrayList<FileLocationStrategy>(Arrays.asList(getSubStrategies())) {
+            @Override
+            public boolean contains(final Object o) {
+                Objects.requireNonNull(o);
+                return super.contains(o);
+            }
+        };
+
+        // This should not throw NPE - the constructor should handle it gracefully
+        final CombinedLocationStrategy strategy = new CombinedLocationStrategy(collectionThatThrowsNPE);
+        checkSubStrategies(strategy);
     }
 
     /**

